internal/hcsoci/credentials.go

```diff
@@ -0,0 +1,115 @@
+// +build windows
+
+package hcsoci
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/Microsoft/hcsshim/internal/hcs"
+	"github.com/Microsoft/hcsshim/internal/log"
+	hcsschema "github.com/Microsoft/hcsshim/internal/schema2"
+	"github.com/sirupsen/logrus"
+)
+
+// CCGState stores the id used when creating a ccg instance. Used when
+// closing a container to be able to release the instance.
+type CCGState struct {
```

Comment:
The "resource" we care about freeing here isn't the state, it's the CCG instance. I would suggest `CCGInstance` or `CCG` as the name. Also please make sure it's clear what a "CCG" is to a new reader in this file.

Reply:
I would probably add a comment at the top of this file explaining what CCG is and how it's used.